xc/xc7/tests/iosettings/CMakeLists.txt

```diff
@@ -58,38 +98,110 @@ function(xc7_iosettings_design MODE BOARD IOSTANDARD DRIVES SLEWS TERMS)
+
 set(BOARD "arty-full")
 
+# Single ended
 foreach(MODE input output inout)
 
+    # Input termination is only relevan for input
     if(MODE STREQUAL "input")
         set(TERMS "NONE UNTUNED_SPLIT_40 UNTUNED_SPLIT_50 UNTUNED_SPLIT_60")
     else()
         set(TERMS "NONE")
     endif()
 
+    # Drive current set #1
     foreach(IOSTANDARD LVCMOS15 LVCMOS25 LVCMOS33)
-        xc7_iosettings_design(${MODE} ${BOARD} ${IOSTANDARD} "4 8 12 16" "SLOW FAST" "NONE")
+        xc7_iosettings_design(
+            MODE ${MODE}
+            BOARD ${BOARD}
+            IOSTANDARD ${IOSTANDARD}
+            DRIVES "4 8 12 16"
+            SLEWS "SLOW FAST"
+            TERMS "NONE"
+        )
     endforeach()
 
+    # Drive current set #2
     foreach(IOSTANDARD LVTTL) # Disabled LVCMOS18 due to #1263 (PUDC_B pin issues)
-        xc7_iosettings_design(${MODE} ${BOARD} ${IOSTANDARD} "4 8 12 16 24" "SLOW FAST" "NONE")
+        xc7_iosettings_design(
+            MODE ${MODE}
+            BOARD ${BOARD}
+            IOSTANDARD ${IOSTANDARD}
+            DRIVES "4 8 12 16 24"
+            SLEWS "SLOW FAST"
+            TERMS "NONE"
+        )
     endforeach()
 
-    xc7_iosettings_design(${MODE} ${BOARD} LVCMOS12 "4 8 12" "SLOW FAST" "NONE")
+    # Drive current set #3
+    xc7_iosettings_design(
+        MODE ${MODE}
+        BOARD ${BOARD}
+        IOSTANDARD LVCMOS12
+        DRIVES "4 8 12"
+        SLEWS "SLOW FAST"
+        TERMS "NONE"
+    )
 
-    # Disabled due to https://github.com/SymbiFlow/prjxray/issues/1182 (unknown bits)
-    #xc7_iosettings_design(${MODE} ${BOARD} SSTL135 "0" "SLOW FAST" ${TERMS})
+    # Fixed drive current
+    foreach(IOSTANDARD SSTL135 SSTL15)
+
+        # VREF is only relevant for input/inout
+        if(NOT MODE STREQUAL "output")
+            foreach(VREF 0.600 0.675 0.750 0.900)
+                xc7_iosettings_design(
+                    MODE ${MODE}
+                    BOARD ${BOARD}
+                    IOSTANDARD ${IOSTANDARD}
+                    DRIVES "0"
+                    SLEWS "SLOW FAST"
+                    TERMS ${TERMS}
+                    VREF ${VREF}
+                )
+            endforeach()
+
+        # Output mode, no VREF setting
+        else()
+            xc7_iosettings_design(
+                MODE ${MODE}
+                BOARD ${BOARD}
+                IOSTANDARD ${IOSTANDARD}
+                DRIVES "0"
+                SLEWS "SLOW FAST"
+                TERMS ${TERMS}
+            )
+        endif()
+    endforeach()
 
 endforeach()
 
-# Currently only one differential setting is supported
-xc7_iosettings_design("diff_output" ${BOARD} DIFF_SSTL135 "0" "SLOW FAST" "NONE")
+# Differential. Currently only differential output (tristated) is supported.
+foreach(IOSTANDARD DIFF_SSTL135 DIFF_SSTL15)
+    xc7_iosettings_design(
+        MODE "diff_output"
+        BOARD ${BOARD}
+        IOSTANDARD ${IOSTANDARD}
+        DRIVES "0"
```

Comment:
Rather than DRIVES "0", maybe DRIVES FIXED?

Reply:
Ok, Now you simply don't give DRIVE argument for IOSTANDARDs that do not require it.